### PR TITLE
Feature/bond already settled transactions

### PR DESCRIPTION
**Pull request by shanefontaine**
**_Do not merge until tests are complete._**

This PR allows the system to bond transfers whose `transferRoots` were already committed/confirmed/settled.

* Marks a transfer as bondable/unbondable upon seeing `TransferSent` event.
* Does not bond if not enough credit is available

With these changes, transfer roots will be settled as many times as needed in order fully "complete" a root hash.

### CODE REVIEW

File: packages/hop-node/src/watchers/classes/Bridge.ts [hunk 1]
```diff
@@ -31,6 +31,8 @@ export default class Bridge extends ContractBase {
   bridgeContract: Contract
   readProvider?: providers.Provider
   bridgeDeployedBlockNumber: number
+  minAmount: BigNumber
```
Comment (miguelmota) [inline packages/hop-node/src/watchers/classes/Bridge.ts:34]: I suggest making these names more descriptive in the context of the bridge; ie `minBondwithdrawalAmount`

File: packages/hop-node/src/watchers/classes/Bridge.ts [hunk 2]
```diff
@@ -67,6 +69,8 @@ export default class Bridge extends ContractBase {
       throw new Error('bridge deployed block number is required')
     }
     this.bridgeDeployedBlockNumber = config.tokens[this.tokenSymbol]?.[this.chainSlug]?.bridgeDeployedBlockNumber
+    this.minAmount = config?.bondWithdrawals?.[this.chainSlug]?.[this.tokenSymbol]?.min ?? 0
```
Comment (miguelmota) [inline packages/hop-node/src/watchers/classes/Bridge.ts:72]: shouldn't these be converted to BigNumber types?

File: packages/hop-node/src/watchers/classes/Bridge.ts (hunk 1 above)
Comment (shanefontaine) [reply packages/hop-node/src/watchers/classes/Bridge.ts]: Resolved with ae73d900762faa1f5898955d07d840716155bc9b

File: packages/hop-node/src/watchers/classes/Bridge.ts (hunk 2 above)
Comment (shanefontaine) [reply packages/hop-node/src/watchers/classes/Bridge.ts]: Resolved with ae73d900762faa1f5898955d07d840716155bc9b

**Pull request merged**